cli/commands/rewards/fetchevents.ts

```diff
@@ -91,9 +93,17 @@ export default class FetchEvents extends BaseCommand {
 
     progressBar.update(progressBarTotal)
     progressBar.stop()
-    const attestationsFile = `attestation-completed-events-${fromBlock}-${toBlock}.json`
-    const transferFile = `transfer-cusd-events-${fromBlock}-${toBlock}.json`
-    this.outputToFile(attestationsFile, attestationEvents, 'AttestationCompleted events')
-    this.outputToFile(transferFile, transferEvents, 'Transfer events')
+    this.writeEventsToFiles(attestationEvents, 'Attestation Events', eventsLimitPerFile)
+    this.writeEventsToFiles(transferEvents, 'Transfer Events', eventsLimitPerFile)
+  }
+
+  writeEventsToFiles = (events: EventLog[], eventTitle: string, limit: number) => {
+    while (events.length > 0) {
+      const splicedEvents = events.splice(0, limit)
+      const fromBlock = splicedEvents[0].blockNumber
+      const toBlock = splicedEvents[splicedEvents.length - 1].blockNumber
```

Comment:
IDK what a good soln here is, but it might be confusing if there's a large gap in blocks between events (eg. 0-5.json 15-22.json). Perhaps we can just index it by the loop count? (0.json 1.json)

Reply:
the gap is unfortunate, yes 🤔...but in terms of caching events for later (so you won't have to fetch until January over and over) block numbers might be nicer.  If I cache, with consecutive numbers I'll get a 0.json, and next time I do it from the next block, I'll also get a 0 json (you _could_ rename it manually). I'm thinking that as someone who's working with and debugging it, the block numbers are a little more informative to work with

Reply:
I think I can include both and have something like `events0-250-2500` events1-2654-40000`